core/src/execution/constants.rs

```diff
@@ -1,5 +1,5 @@
 pub const PARALLEL_QUERY_BATCH_SIZE: usize = 20;
 
-// We currently limit the max number of logs to fetch,
+// We currently limit the max number of blocks to prove a list of logs,
 // to avoid blocking the client for too long.
-pub const MAX_SUPPORTED_LOGS_NUMBER: usize = 5;
+pub const MAX_SUPPORTED_BLOCKS_TO_PROVE_FOR_LOGS: usize = 5;
```

Comment:
What do you think the right number here is?

Reply:
It might also make sense to have different values here for the API vs RPC since the overheads are pretty different for these. The API should be able to handle a higher value.

Reply:
Having different values for API vs RPC does make sense. Might also wanna do it for `PARALLEL_QUERY_BATCH_SIZE`.

As for what is the right number, I think we need to make this value configurable by the end-user. For example, somebody using a dedicated RPC could afford a lot more requests.

Reply:
Implemented endpoints for these methods in 236a9052564f2832bc0d6780071538511cb4657a.